include/zephyr/net/wifi_nm_wpa_supp.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_WIFI_NM_WPA_SUPP_H_
+#define ZEPHYR_INCLUDE_NET_WIFI_NM_WPA_SUPP_H_
+
+#include <zephyr/net/wifi_nm.h>
+
+union wifi_nm_supp_mon_event_data {
+	struct wifi_nm_supp_event_auth_reject {
+		int auth_type;
+		int auth_transaction;
+		int status_code;
+		uint8_t bssid[ETH_ALEN];
+	} auth_reject;
+
+	struct wifi_nm_supp_event_connected {
+		uint8_t bssid[ETH_ALEN];
+		char ssid[SSID_MAX_LEN];
+		int id;
+	} connected;
+
+	struct wifi_nm_supp_event_disconnected {
+		uint8_t bssid[ETH_ALEN];
+		int reason_code;
+		int locally_generated;
+	} disconnected;
+
+	struct wifi_nm_supp_event_assoc_reject {
+		int status_code;
+		int reason_code;
+	} assoc_reject;
+
+	struct wifi_nm_supp_event_temp_disabled {
+		int id;
+		char ssid[SSID_MAX_LEN];
+		unsigned int auth_failures;
+		unsigned int duration;
+		char reason_code[32];
+	} temp_disabled;
+
+	struct wifi_nm_supp_event_reenabled {
+		int id;
+		char ssid[SSID_MAX_LEN];
+	} reenabled;
+
+	struct wifi_nm_supp_event_bss_added {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_added;
+
+	struct wifi_nm_supp_event_bss_removed {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_removed;
+
+	struct wifi_nm_supp_event_network_added {
+		unsigned int id;
+	} network_added;
+
+	struct wifi_nm_supp_event_network_removed {
+		unsigned int id;
+	} network_removed;
+
+	char wifi_nm_supp_event_str[64];
+};
+
+enum wifi_nm_supp_event_num {
+	CONNECTED,
+	DISCONNECTED,
+	ASSOC_REJECT,
+	AUTH_REJECT,
+	TERMINATING,
+	SSID_TEMP_DISABLED,
+	SSID_REENABLED,
+	SCAN_STARTED,
+	SCAN_RESULTS,
+	SCAN_FAILED,
+	BSS_ADDED,
+	BSS_REMOVED,
+	NETWORK_NOT_FOUND,
+	NETWORK_ADDED,
+	NETWORK_REMOVED
+};
```

Comment:
Are these values used in the `event` field in `struct wifi_nm_supp_status_data`? If yes, then please change the type of the said event field to `enum  wifi_nm_supp_event_num` so the relation is clear.

Reply:
Also please namespace the enums so that they do not conflict with other code, now the names are too generic.

Reply:
Done.